test/test_jit.py

```diff
@@ -1219,30 +1219,33 @@ def allSum(vs):
         # test single grad case
 
         outputs = func(*recording_inputs)
-        grads = torch.autograd.grad(allSum(outputs), recording_inputs)
+        grads = torch.autograd.grad(allSum(outputs), recording_inputs,
+                                    allow_unused=True)
 
         outputs_ge = ge(*recording_inputs)
-        grads_ge = torch.autograd.grad(allSum(outputs_ge), recording_inputs)
+        grads_ge = torch.autograd.grad(allSum(outputs_ge), recording_inputs,
+                                       allow_unused=True)
         self.assertEqual(outputs, outputs_ge)
         self.assertEqual(grads, grads_ge)
 
         # test the grad grad case
 
         outputs = func(*recording_inputs)
         l1 = allSum(outputs)
-        grads = torch.autograd.grad(l1, recording_inputs, create_graph=True)
+        grads = torch.autograd.grad(l1, recording_inputs, create_graph=True,
+                                    allow_unused=True)
         l2 = (allSum(grads) * l1)
-        grads2 = torch.autograd.grad(l2, recording_inputs)
+        grads2 = torch.autograd.grad(l2, recording_inputs, allow_unused=True)
 
         recording_inputs = [Variable(t, requires_grad=True)
                             for t in reference_tensors]
 
         outputs_ge = ge(*recording_inputs)
         l1_ge = allSum(outputs_ge)
         grads_ge = torch.autograd.grad(
-            l1_ge, recording_inputs, create_graph=True)
+            l1_ge, recording_inputs, create_graph=True, allow_unused=True)
         l2_ge = (allSum(grads_ge) * l1_ge)
-        grads2_ge = torch.autograd.grad(l2_ge, recording_inputs)
+        grads2_ge = torch.autograd.grad(l2_ge, recording_inputs, allow_unused=True)
```

Comment:
Adding `allow_unused` makes those tests much less effective. We need to change the `func` to have a non-zero grad here

Reply:
The new commit changes it such that we only set `allow_unused` with the `func` that has unused input.